src/Domain/TagsCharacteresticSpeeds.hpp

```diff
@@ -0,0 +1,54 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <boost/optional.hpp>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"  // For Tags::Normalized
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Tags.hpp"
+#include "Domain/TagsTimeDependent.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace domain {
+namespace Tags {
+/// Compute the characteristic speeds on the moving mesh given the
+/// characteristic speeds if the mesh were stationary.
+///
+/// \note Assumes that `typename CharSpeedsComputeTag::return_type` is a
+/// `std::array<DataVector, NumCharSpeeds>`
```

Comment:
I suggest naming the second template parameter in the dox NumberOfCharSpeeds, to match the code